### PR TITLE
[MM-46]: Updated the message for exporting channels

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
#### Summary
Updated the messages received while exporting different types of channels

#### Existing
Screenshot from 2024-08-14 09-19-55
Screenshot from 2024-08-14 09-05-19
Screenshot from 2024-08-21 14-16-56
Screenshot from 2024-08-21 14-17-40

#### Updated
Screenshot from 2024-08-14 09-06-30
Screenshot from 2024-08-21 14-16-20
Screenshot from 2024-08-21 16-30-46
Screenshot from 2024-08-14 09-05-51

### CODE REVIEW

**Comment by raghavaggarwal2308** [comment]
@Kshitij-Katiyar In `Exporting GM channel ~ID`. Shouldn't "~ID" be a link?

**Comment by Kshitij-Katiyar** [comment]
> @Kshitij-Katiyar In `Exporting GM channel ~ID`. Shouldn't "~ID" be a link?

@raghavaggarwal2308 This Can be a link, but not much needed as the given message is received in the GM channel only, but can be made a link to maintain the consistency. Should we?

**Comment by raghavaggarwal2308** [comment]
> > @Kshitij-Katiyar In `Exporting GM channel ~ID`. Shouldn't "~ID" be a link?
> 
> @raghavaggarwal2308 This Can be a link, but not much needed as the given message is received in the GM channel only, but can be made a link to maintain the consistency. Should we?

Yes, I think we should. Just for consistency

**Comment by Kshitij-Katiyar** [comment]
> > > @Kshitij-Katiyar In `Exporting GM channel ~ID`. Shouldn't "~ID" be a link?
> > 
> > 
> > @raghavaggarwal2308 This Can be a link, but not much needed as the given message is received in the GM channel only, but can be made a link to maintain the consistency. Should we?
> 
> Yes, I think we should. Just for consistency

@raghavaggarwal2308 We have implemented your suggestions and updated the screenshot accordingly. Please have a look

File: server/command_hooks.go [hunk 1]
```diff
@@ -88,6 +88,35 @@ func (p *Plugin) executeCommandExport(args *model.CommandArgs) *model.CommandRes
...
+	if channelToExport.Type == model.ChannelTypeOpen || channelToExport.Type == model.ChannelTypePrivate {
+		channelToExportName = channelToExport.Name
+		exportSuccessMsg = fmt.Sprintf("Channel ~%s exported:", channelToExportName)
+		exportProcessMsg = fmt.Sprintf("Exporting ~%s. @%s will send you a direct message when the export is ready.", channelToExportName, botUsername)
+	} else if channelToExport.Type == model.ChannelTypeGroup {
+		channelToExportName = channelToExport.Name
+		team, err := p.API.GetTeam(args.TeamId)
+		if err != nil {
+			p.client.Log.Error("error occurred while getting the details of team for the channel.",
+				"GM channel ID", args.ChannelId, "User ID", args.UserId, "Error", err)
```
Comment (raghavaggarwal2308) [inline server/command_hooks.go]:
```suggestion
			p.client.Log.Error("error occurred while getting the details of team for the channel.", "GMChannelID", args.ChannelId, "TeamID", args.TeamId, "Error", err)
```

File: server/command_hooks.go [hunk 2]
```diff
@@ -88,6 +88,35 @@ func (p *Plugin) executeCommandExport(args *model.CommandArgs) *model.CommandRes
...
+
+			return &model.CommandResponse{
+				ResponseType: model.CommandResponseTypeEphemeral,
+				Text:         fmt.Sprintf("An error occurred trying to exporting the chat for %s GM channel", channelToExportName),
+			}
```
Comment (raghavaggarwal2308) [inline server/command_hooks.go]: Having this link is not a must so we can just log the error and continue with our functionality without a link

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/command_hooks.go [hunk 3]
```diff
@@ -88,6 +88,48 @@ func (p *Plugin) executeCommandExport(args *model.CommandArgs) *model.CommandRes
...
 
+	var channelToExportName, exportSuccessMsg, exportProcessMsg string
+	switch channelToExport.Type {
+	case model.ChannelTypeOpen, model.ChannelTypePrivate:
+		channelToExportName = channelToExport.Name
+		exportSuccessMsg = fmt.Sprintf("Channel ~%s exported:", channelToExportName)
+		exportProcessMsg = fmt.Sprintf("Exporting ~%s. @%s will send you a direct message when the export is ready.", channelToExportName, botUsername)
+	case model.ChannelTypeGroup:
+		channelToExportName = channelToExport.Name
+		team, appErr := p.API.GetTeam(args.TeamId)
```
Comment (raghavaggarwal2308) [inline server/command_hooks.go:99]: use p.client

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/command_hooks.go (hunk 3 above)
Comment (raghavaggarwal2308) [reply server/command_hooks.go:99]: discussed

**Comment by Kshitij-Katiyar** [comment]
Created a PR with all the changes on mattermost repo, hence closing this one.

**Pull request closed without merging**